src/zenml/steps/base_step.py

```diff
@@ -291,37 +291,87 @@ def _prepare_parameter_spec(self) -> None:
                     f"json serializable parameter values."
                 ) from e
 
+    def _prepare_input_artifacts(
+        self, *artifacts: Channel, **kw_artifacts: Channel
+    ) -> Dict[str, Channel]:
+        """Verifies and prepares the input artifacts for running this step.
+
+        Args:
+            *artifacts: Positional input artifacts passed to
+                the __call__ method.
+            **kw_artifacts: Keyword input artifacts passed to
+                the __call__ method.
+
+        Returns:
+            Dictionary containing both the positional and keyword input
+            artifacts.
+
+        Raises:
+            StepInterfaceError: If there are too many or too few artifacts.
+        """
+        input_artifact_keys = list(self.INPUT_SPEC.keys())
+        if len(artifacts) > len(input_artifact_keys):
+            raise StepInterfaceError(
+                f"Too many input artifacts for step '{self.step_name}'. "
```

Comment:
Cool one! That was weird in the zenhack

Reply:
Yeah that was my motivation to quickly fix this as some of the errors with args/kwargs and strage error messages happened during zenhack